collector/builder-config.yaml

```diff
@@ -106,11 +106,19 @@ connectors:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector v0.106.0
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector v0.106.0
 
+providers:
+  - gomod: go.opentelemetry.io/collector/odigos/providers/odigosfileprovider v0.106.0 # fork default file provider for config reloading
+  - gomod: go.opentelemetry.io/collector/confmap/provider/envprovider v0.106.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpprovider v0.106.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpsprovider v0.106.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/yamlprovider v0.106.0
```

Comment:
Why are those needed?

Reply:
See https://github.com/odigos-io/odigos/pull/1903#discussion_r1870208029

Reply:
I think we can remove them now that we have a provider and we don't use them.

Reply:
We probably will still want the env provider at least, others I think can be removed